### DISCUSSION TITLE
Possible to setup photoprism all behind a firewall on desktop?

### DISCUSSION THREAD

**Discussion by greereer** [photoprism/photoprism#3350] [General]
Hello,

Is this possible to setup the server, database and client on one desktop pc and have it all communicate 'behind the firewall'? I have tried setting up a photoprism docker container but the browser client still needs to go past the portmaster firewall on my desktop so it can accesses the server/database.
Thanks for your help please advise if there is a better way of putting my question
Greer

**Comment by lastzero** [comment]
Yes, it's definitely possible. You're probably using Windows then? I've never head of a "portmaster" firewall before 🤔

**Reply by greereer** [reply]
The portmaster is a graphical firewall and I am.... trying to use this all on linux...... this has been interesting

**Reply by lastzero** [reply]
I would then recommend ufw ("Uncomplicated Firewall"):

```bash
apt-get update
apt-get install ufw

ufw default deny incoming
ufw default allow outgoing
ufw allow ssh
ufw allow http
ufw allow https
ufw allow 2342
ufw logging off
ufw --force enable
```
